pkg/server/server.go

```diff
@@ -178,6 +180,163 @@ func NewServer(c CompletedConfig) (*Server, error) {
 	return s, nil
 }
 
+/* Deregister the controllers when the leader election is lost. */
+func (s *Server) uninstallControllers() {
```

Comment:
these run under the lock. `uninstallControllersLocked()` is typical style to make that clear.

Reply:
removed the lock as suggested above